src/ripple/ledger/impl/PaymentSandbox.cpp

```diff
@@ -110,19 +133,29 @@ DeferredCredits::adjustments (AccountID const& main,
 void DeferredCredits::apply(
     DeferredCredits& to)
 {
-    for (auto& p : map_)
+    for (auto const& i : map_)
     {
-        auto r =
-            to.map_.emplace(p);
-        if (! r.second)
+        auto r = to.map_.emplace (i);
+        if (!r.second)
         {
             auto& toVal = r.first->second;
-            auto const& fromVal = p.second;
+            auto const& fromVal = i.second;
             toVal.lowAcctCredits += fromVal.lowAcctCredits;
             toVal.highAcctCredits += fromVal.highAcctCredits;
             // Do not update the orig balance, it's already correct
         }
     }
+
+    for (auto const& i : ownerCounts_)
+    {
+        auto r = to.ownerCounts_.emplace (i);
+        if (!r.second)
+        {
+            auto& toVal = r.first->second;
+            auto const& fromVal = i.second;
+            toVal = std::max (toVal, fromVal);
```

Comment:
How do we know that `toVal` only increases?  Not saying it's wrong, but it's a bit counterintuitive to me.  A comment with the justification would be nice.